### DISCUSSION TITLE
小米手机无法正常使用

### DISCUSSION THREAD

**Discussion by 100730088** [kkkgo/PaoPaoGateWay#34] [问和答]
![image](https://github.com/kkkgo/PaoPaoGateWay/assets/3375457/f59026b2-4204-4312-8f7b-4d8a7882784f)
小米手机上把网关和DNS都设为PPGW，国内的网站和手机APP可以可以正常使用，但是其他的网站打不开，查看PPGW的网页，发现HOST全是IP。
电脑上同样的设置，就都是正常的，PPGW的WEB上，HOST是域名。

**Comment by kkkgo** [comment]
把网关和DNS都设为PPGW相当于全局走PPGW哦，你确定是要这么用嘛……
你手机可能设置了安全DNS之类的吧，没用到FAKEIP。

**Reply by kkkgo** [reply]
另外网站打不开大概和你的线路/规则有关，即使没用使用到FAKE DNS，流量也是能正常出站的。

**Reply by 100730088** [reply]
目前还在尝试，所以不大想改ROS主路由的设置。
PPGW用的是SOCKS5连的，在后面做了规则。
规则是基于域名的吧，所以HOST是IP的话，后面就都 MATCH,DIRECT 直连了。

手机是小米的，找了半天并没有设置安全DNS的地方。
非常奇怪的问题

**Reply by kkkgo** [reply]
https://jingyan.baidu.com/article/3c48dd34b4533ca00ae3583b.html

**Reply by 100730088** [reply]
看了下是关闭的。
我的ROS还开了IPV6,会是这个的影响吗。

另外有ROS PPGW PPDNS配合使用的教程吗。
我对主路由DNS、路由表、DHCP设置并不是很清楚

**Reply by kkkgo** [reply]
如果你下发了IPV6的DNS地址，客户端一般会优先用IPV6 DNS。
客户端在域名取得AAAA记录会优先用IPV6地址访问，不走PPGW网关，那么你无法访问就说得通了。
配置可以参考: https://github.com/kkkgo/PaoPaoGateWay/discussions/categories/%E9%85%8D%E7%BD%AE%E5%88%86%E4%BA%AB

**Reply by kkkgo** [reply]
使用IPV6的话，可以让主路由不下发IPV6的DNS，只使用ppdns的dns就可以了。ppdns需要转发的域名过去ppgw就不会有AAAA记录。

**Comment by hsj1992** [comment]
还是不建议这样使用PaoPaoGateWay，你客户端直接指向它的话，相当于把分流任务交给它里头的Clash，就靠规则模式的规则来分流了。
这样的话就没什么“fakeip旁路网关”了，不如直接用clash。
android的设备有个问题，哪怕你配置了ipv4，有ipv6的情况下会优先使用ipv6的dns解析。这也是为什么你的android设备无法使用PaoPaoGateWay。
所以你可以：
思路1：使用ULA，然后ipv6也指向PaoPaoGateWay的ULA地址。
思路2：关闭、不使用ipv6。
思路3：主路由不下发ipv6的DNS。
思路4：使android设备无法使用ipv6 DNS从而使用ipv4的PaoPaoGateWay。

**Comment by LetMeDecay** [comment]
让主路由下发一个无效的v6 dns就行了，这样只有手动配置了dns的设备才会走网关，其他设备默认还是走路由器
[https://www.right.com.cn/forum/forum.php?mod=redirect&goto=findpost&ptid=5418059&pid=15346559](https://www.right.com.cn/forum/forum.php?mod=redirect&goto=findpost&ptid=5418059&pid=15346559)
